### PR TITLE
fix(e2ei): loading e2ei state during the app initialisation

### PR DESCRIPTION
**Pull request by mchenani**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
Loading the e2ei state if it's blocking the client registration during login when reopening the app as a blocking value.

### Issues

The value of e2ei state was not loaded in blocking state, causes incorrect behaviour.

#### How to Test

Login with an account that has E2EI enabled on their team, then while getting the E2EI certificate, kill and re open the app; the app must open the enrol certificate screen again to continue from there.
----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2885](https://10.10.124.134/job/AR-build-pipeline/2885/) **failed**.

**Review by saleniuk — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt [hunk 1]
```diff
@@ -434,10 +425,11 @@ class WireActivityViewModel @Inject constructor(
 
     fun shouldLogIn(): Boolean = !hasValidCurrentSession()
 
-    fun blockedByE2EI(): Boolean {
-        return observeE2EIState.value == true
+    private fun blockedByE2EI(): Boolean = runBlocking {
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt:428]: question: do we need it to runBlocking can't we have it return a `Deferred` and await the result when needed ?

**Comment by AndroidBob** [comment]
Build [3354](https://10.10.124.134/job/AR-build-pipeline/3354/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt (hunk 1 above)
Comment (mchenani) [reply app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt:428]: We reviewed online and it's resolved.

**Review by MohamadJaara — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [3370](https://10.10.124.134/job/AR-build-pipeline/3370/) **failed**.

**Comment by AndroidBob** [comment]
Build [3414](https://10.10.124.134/job/AR-build-pipeline/3414/) **failed**.

**Comment by AndroidBob** [comment]
Build [3415](https://10.10.124.134/job/AR-build-pipeline/3415/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.6.1(9030674)(9030674)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.6.1(9030674)(9030674)-staging-compat.apk)

**Pull request merged**